### ISSUE TITLE
Videos resized to square on iOS

### ISSUE THREAD

**Issue by amiraria4** [HirbodBehnam/RedditDownloaderBot#12]
Videos larger than 10 MB are resized to square, regardless of their orientation.

Only on Telegram for iOS. Android and Desktop versions display videos normally.

Example: https://www.reddit.com/r/funny/comments/ysrihs/average_twitter_user

On iOS:
![iOS](https://user-images.githubusercontent.com/78235035/201517427-aed1132d-a8c9-46dd-a089-595980229566.jpg)

On Desktop:
![Desktop](https://user-images.githubusercontent.com/78235035/201517431-1e154a61-5f28-4f78-9c07-dfaa75185f41.png)

**Comment by amiraria4** [comment]
Actually, this is the case for most Telegram bots, fetching and reuploading videos that are larger than 10 MB. Whether they are horizontal or vertical, they will be displayed as square on Telegram for iOS. (Another example is [utuberabot](https://t.me/utuberabot)). However, I doubt that this is a bug with Telegram, as, if I upload a 10+ MB video on iOS, it will be displayed normally.

Can you please look into this?

**Comment by HirbodBehnam** [comment]
Does this still happen? This looks a like an issue on Telegram side because as you said, this is the case with most bots and I do not have such an issue on Android and Desktop.

**Comment by HirbodBehnam** [comment]
I just realized that Telegram accepts width and height of photos and videos as the parameters. I think I can probably fill them out before sending the file to Telegram. I'll try to write something about them later after I have merged #26

**Comment by mcsaeid** [comment]
This is and has been unique to iOS, which is unfortunate. Like @amiraria4 said, a user-uploaded video, regardless of its size and dimension, will display correctly across all platforms. It’s only when bots upload videos that this problem occurs, and only on iOS devices. Granted, this doesn’t happen with _all_ bots, but when it does, it’s annoying.

A workaround I’ve found, albeit a little time-consuming, is reuploading the video using another bot, [URL Uploader X](https://t.me/urluploadxbot). Take a 10+ MB Reddit video, fetched by your bot and plagued with this bug, if you will. If I send it to the uploader bot and simply rename it so that it will download and then upload the video again, it will be shown in the correct aspect ratio. I’m not a technical person, but clearly, there’s something happening on the Telegram side, perhaps pertaining to the API, that is causing this issue. And given your last comment, I believe you’re on the right track. I hope you’re able to fix this.

By the way, I forked the project [here](https://github.com/mcsaeid/RedditDownloaderBot), wrote a more extensive Readme that includes pictures (since the original installation guide wasn’t quite intuitive for someone like me, with very little programming skills), and also went through the codebase and made sure every text that’s ever displayed to the end-user is grammatically and punctually correct. (Call it a pet peeve of mine.) Feel free to use any part of the fork however you see fit.

Thank you so much for your awesome work and continued support. Have a good one!

**Comment by HirbodBehnam** [comment]
@mcsaeid 
Thanks for the idea. I don't think using another bot with conjunction of mine is a good idea :)) But maybe that bot uses a [Local Bot API Server](https://core.telegram.org/bots/api#using-a-local-bot-api-server) and that avoids it. Regardless, I'll try to at first try with the dimensions then I'll try to do some fallbacks.

By the way, can you open a pull request from your fork to the `gotgbot` (and not the master!) branch?

Much appreciated for your work!

**Comment by HirbodBehnam** [comment]
I've actually made [this](https://github.com/HirbodBehnam/RedditDownloaderBot/pull/26/commits/8c3fd50d899da574c47335d612447cd4bce044dc) change to get the dimension of videos before sending them to Telegram. Can anyone check and see if it fixes the problem?

**Comment by mcsaeid** [comment]
I updated the bot and tested it on my iOS device. Unfortunately, the issue persists. Here are my findings. Let me know what you think.

![Image](https://github.com/user-attachments/assets/fa6dee58-0218-4783-9fae-f59d0b640177)

Also, take a look at this [bug report](https://github.com/TelegramMessenger/Telegram-iOS/issues/1477) on the Telegram iOS repository. The OP may have solved this problem.

**Comment by mcsaeid** [comment]
**Update**
Weirdly, the bug seems to be gone. I have no clue how. It may have been the recent update to the code, or maybe it was something on the Telegram side, but I tested it on three separate iOS devices, and they all played videos correctly. No more squared videos. Only one issue remains now. For whatever reason, we still don’t get thumbnail on files over 10 megabytes. Maybe you can focus on finding a fix for this.

**Comment by HirbodBehnam** [comment]
I've also realized that the thumbnails are broken. Apparently, they need authentication to be downloaded (unlike other media). This requires some rewrite because the downloaders should also have access to the Reddit ouath. I'll figure something out...

**Comment by HirbodBehnam** [comment]
I've wrote a scratch of it here:
https://github.com/HirbodBehnam/RedditDownloaderBot/pull/26/commits/d6cde2e43ffb992329160109eeebeb985495b9a7
Can anyone test it to see if it's fine or not?

**Comment by HirbodBehnam** [comment]
I've also made some changes in https://github.com/HirbodBehnam/RedditDownloaderBot/commit/e32b05436819ba1d2c5910c91df1d197de7760eb to restrict the size of thumbnail to 1080x1080. From the Telegram official documentation, the maximum dimension of a thumbnail can be 320x320 which looks like wrong to me because I've tested with larger images and they were fine. I think the file size of 200KB is the limit here not the dimension.
Regardless, with this commit it should be easier to change the maximum size of a thumbnail or remote it completely.

I think I'm also going to roll out a public beta test on my bot to see how it handles stuff...
